test/integration/builder/triggers.js

```diff
@@ -0,0 +1,202 @@
+
+        // Create proper environment for tests
+        before(async function () {
+            if (knex.client.driverName !== 'mssql') {
+                this.skip();
+            }
+
+            await knex.schema.hasTable('users').then(async function (exists) {
+                if (exists) {
+                    await knex.schema.dropTable(primaryTable);
+                    await knex.schema.dropTable(secondaryTable);
+                }
+
+                // Create tables
+                await knex.schema.createTable(primaryTable, function (table) {
+                    table.increments();
+                    table.string("data");
+                    table.integer(primaryLink);
+                });
+
+                await knex.schema.createTable(secondaryTable, function (table) {
+                    table.increments();
+                    table.string("data");
+                    table.integer(secondaryLink);
+                });
+            });
+        });
+
+        // Clean-up test specific tables
+        after(async function () {
+            if (knex.client.driverName !== 'mssql') {
+                return;
+            }
+
+            // Drop table (Trigger is removed with table)
+            await knex.schema.dropTable(primaryTable);
+            await knex.schema.dropTable(secondaryTable);
+        });
+
+        // Reset tables for each test
+        beforeEach(async function () {
+            if (knex.client.driverName !== 'mssql') {
+                return;
+            }
+
+            // "Truncate" tables instead of recreate for each test for speed gains
+            await knex.raw(`
+                delete from ${primaryTable} dbcc checkident('${primaryTable}', reseed, 0);
+                delete from ${secondaryTable} dbcc checkident('${secondaryTable}', reseed, 0);
+            `);
+        })
+
+        describe("Inserts with Triggers", function () {
+            // Trigger Name
+            const triggerName = 'tr_test_insert_with_trigger';
+
+            before(async function () {
+                if (knex.client.driverName !== 'mssql') {
+                    return;
+                }
+
+                // Create trigger
+                await knex.raw(`
+                    CREATE TRIGGER [${triggerName}] ON [${secondaryTable}]
+                    AFTER INSERT
+                    AS 
+                    BEGIN
+                        SET NOCOUNT ON;
+
+                        BEGIN
+                            update pt
+                            set pt.${primaryLink} = i.id
+                            from Inserted as i
+                            inner join ${primaryTable} as pt
+                                on pt.id = i.${secondaryLink}
+                        END
+                    END
+                `);
+            });
+
+            it('#4152 Should allow returns with inserts on tables with triggers', async function () {
+                if (knex.client.driverName !== 'mssql') {
```

Comment:
if `this.skip` is executed in `before`, it shouldn't get here, so it shouldn't be necessary.